torch/utils/data/_utils/worker.py

```diff
@@ -223,6 +223,14 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
         seed = base_seed + worker_id
         random.seed(seed)
         torch.manual_seed(seed)
+        for generator in torch.random._generator_registry:
+            # We have to reseed non-global Generators as well (TODO: explain why, will do if PR is likely to be merged)
+            # Note that we can't use generator.manual_seed(seed) here, otherwise the RNG of the generator
+            # would be the same as the global one.
+            # We also need to create a generator_seed that depends on the current generator state, otherwise
+            # all Generator instances within a given worker would have the same RNG.
+            generator_seed = torch.empty((), dtype=torch.int64).random_(generator=generator).item() + seed
+            generator.manual_seed(generator_seed)
```

Comment:
Technically this is a change of behaviour: In `main`, the `Generator` instances have the same RNG across workers; now they don't.

I don't think users are / should rely on this behaviour anyway? The only use I can think of where users would *want* to have the same RNG across workers is if they're using the Generator to shuffle the datasets: when shuffling, you want all workers to shuffle in the same way. But for those who need that, I think it's fair to say that they should be relying on `worker_init_fn` anyway.

Reply:
This is technically BC-breaking so we should properly document it (please add a small paragraph in the description for the release notes). But that's ok to do it I think yes.